woocommerce/woocommerce
# Marketplace search view more button & categories

### Changes proposed in this Pull Request:

Closes https://github.com/Automattic/woocommerce.com/issues/18213

- Add "View All" button to themes and extension on search page
- When clicked, the button will load only those results with the category selector shown
- Fixes some bugs with URL query parameters not fully being reset when switching tabs/pages
- Consolidates the `Extensions` and `Themes` components into a single `Products` component
- Move some logic around between components to make sure everything can be dynamic
- Keeps all the search functionality (switching product types etc.) on a single tab to prevent an additional API request.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Checkout this branch.
2. Build WooCommerce Admin on your local environment: `cd plugins/woocommerce-admin` then `nvm use && pnpm build`
3. Visit WooCommerce > Extensions (e.g. http://localhost:8888/wp-admin/admin.php?page=wc-admin&path=%2Fextensions&tab=extensions)
4. Perform a search for `store` – this will give some good results to compare
5. See that the page loads with a search tab and that there are “view all” buttons showing below both Extensions and Themes results
6. See that no category selector appears on the all results page
7. Click on either of the “View All” buttons and see that the results for that type are loaded correctly
8. See that the correct category selector appears
9. See that you should also be able to navigate back using browser history
10. See that after clicking the “View All” button for extensions, you can navigate to the “Themes” tab directly (previously this would display the incorrect product images)
11. Test around this issue



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
